### ISSUE TITLE
[UX][A11Y] Visually indicate a View Display is disabled

### ISSUE THREAD

**Issue by laryn** [backdrop/backdrop-issues#4293]
The main content of the view display gets greyed out if it is disabled, but the main link to the display does not. I think it makes sense to carry that through to the display links at the top like this:

![Screen Shot 2020-01-24 at 11 27 48 AM](https://user-images.githubusercontent.com/871421/73089777-942cb680-3e9c-11ea-954c-94e076eb1912.jpg)

---

~~PR: https://github.com/backdrop/backdrop/pull/3049~~
Alt PR: https://github.com/backdrop/backdrop/pull/3058

**Comment by indigoxela** [comment]
Good catch and it's a minimal change.

**Comment by BWPanda** [comment]
My only concern is that people may not realise they can click on the disabled view display to rehenable it...

**Comment by jenlampton** [comment]
We also need to check that contrast for accessibility. We might want to do a ghost button instead.

**Comment by jenlampton** [comment]
Yeah, there's not enough contrast between the button background and the text color: 
<img width="373" alt="Screen Shot 2020-01-28 at 5 25 01 PM" src="https://user-images.githubusercontent.com/397895/73320124-33321500-41f3-11ea-936f-51aa718de896.png">

**Comment by jenlampton** [comment]
Well, it looks the contrast on the rest of the disabled UI is not readable either... Should we fix both in this issue, or create another? 
<img width="1465" alt="Screen Shot 2020-01-28 at 5 30 23 PM" src="https://user-images.githubusercontent.com/397895/73320386-fc103380-41f3-11ea-854a-9a291eab05c7.png">

**Comment by klonos** [comment]
How about indicating disabled displays by making the "This display is disabled" text a warning at the top of the page, instead of using opacity? That way, we don't have to worry about readability. Then we could be appending `(disabled)` in the display link as well.

I think that we should also indicate disabled displays in the main views listing page.

Something like this for example?:

![Screen Shot 2020-02-02 at 11 42 30 am](https://user-images.githubusercontent.com/2423362/73601545-79062900-45b7-11ea-9837-60861881c2da.png)

![Screen Shot 2020-02-02 at 11 57 24 am](https://user-images.githubusercontent.com/2423362/73601546-7c99b000-45b7-11ea-8915-43ec587834b6.png)

PR: https://github.com/backdrop/backdrop/pull/3058

**Comment by olafgrabienski** [comment]
> How about indicating disabled displays by making the "This display is disabled" text a warning at the top of the page, instead of using opacity?

Not sure but an Interesting idea! If we choose this approach, I'd prefer a blue info message instead of a warning to not confuse it with the "All changes are stored temporarily" warning which appears as soon as you start to configure a view. In any case, for a disabled display it should be sufficient to just inform about it.

**Comment by klonos** [comment]
Yup, it indeed feels better with the message being of type `info` instead of `warning`. Great suggestion @olafgrabienski 👍

**Comment by olafgrabienski** [comment]
Thanks for updating the PR, @klonos. As we have two PRs now, I've added the `needs - more feedback` label to the issue. I'm still not sure which approach to prefer. However, if we choose your PR, I'd suggest to display the name of a disabled display as 'line-through' (see screenshot mockup at bottom).

I've also tested @klonos' PR in the sandbox site. It works fine but when I saved a view, the info message was rendered twice. Speaking of the info message, I'd prefer to not show the info at the top but below the warning message, so that the info is placed directly above the **Displays** heading when there are other messages.

**Screenshot mockup: strike-through for name of disabled display**

![screen-backdrop-pr-3058-b-linethrough](https://user-images.githubusercontent.com/14188307/73638429-ab249300-466a-11ea-8177-445fa38bc266.png)

**Comment by klonos** [comment]
> I'd suggest to display the name of a disabled display as 'line-through' 

I'm neither hot nor cold re this; let's wait for some more feedback before I update the PR.

> when I saved a view, the info message was rendered twice.

Ah yes, this does happen sometimes. Usually the solution is to add a "blank" `backdrop_set_message()` before the actual `backdrop_set_message(t('bla bla bla'))`. I'll test locally and update the PR if that indeed fixes the issue.

> I'd prefer to not show the info at the top but below the warning message

Hmm 🤔 I don't think that you can control the order of messages thrown by `backdrop_set_message()` (although that'd be nice), but I have an idea re this issue: we can use the same approach as in #3330 (but we should really at some point implement #3329 in the Form API level). ...I've updated the PR to implement it that way. Here's current behavior:

![Screen Shot 2020-02-03 at 11 07 29 pm](https://user-images.githubusercontent.com/2423362/73652220-7a7c4200-46da-11ea-9cdf-12521dbcd467.png)

...and here's a screenshot from the updated PR (with the "dirty form" message):

![Screen Shot 2020-02-03 at 11 10 42 pm](https://user-images.githubusercontent.com/2423362/73652244-8bc54e80-46da-11ea-96d6-bd250a958c91.png)

**Comment by BWPanda** [comment]
Is the duplicated messages related to https://github.com/backdrop/backdrop-issues/issues/4247?

**Comment by klonos** [comment]
Yes @BWPanda, it is. I've missed that issue, so thanks for pointing it out 👍

**Comment by laryn** [comment]
I like this (and agree we might as well fix the grey-out accessibility issue right in this issue). The info message is great, and adding "(disabled)" to the display name works for me. I don't have a strong opinion on whether to also add a lineout to the name.

**Comment by NormPlum** [comment]
I came here to make an issue for this, and found an existing one instead.

Looks like nothing much has happened here recently, so I made a new PR that just does the absolute minimum. I don't think adding a strikethrough for disabled displays is a good idea as that's already used for deleted displays:

<img width="487" height="159" alt="Image" src="https://github.com/user-attachments/assets/cb018bfb-6cfa-4e5e-a42d-ad53f226293f" />

(the Attachment display was deleted and shows a strikethrough before the View is saved)

The other PR changed the title of the display such that the '(disabled)' text was shown everywhere (including the drop-down on the right). My PR only changes the titles of the tabs themselves:

<img width="1202" height="342" alt="Image" src="https://github.com/user-attachments/assets/843955d9-8601-46db-8712-55cd87510225" />

**Comment by olafgrabienski** [comment]
@NormPlum Thanks for reviving this issue and providing a minimal approach PR! I've tested the PR in the sandbox, and it works for me.

If I recall the conversation correctly, we agreed to also fix the grey-out accessibility problem in this issue here. Are you up to include something like [this fix](https://github.com/backdrop/backdrop/pull/3058/changes#diff-21aeae88b077af36aee74c2a18265095d1e72c7a67d37c7eea7e37890097cf01L129) in your PR?

**Comment by NormPlum** [comment]
@olafgrabienski I think the accessibility problem requires a dedicated issue and discussion, as the same thing happens on the Views list page:

<img width="1211" height="301" alt="Image" src="https://github.com/user-attachments/assets/6e8e4a4f-4e1d-4815-9290-d10ac38b547d" />

.
If we make disabled views/displays appear with full opacity, we'll need another way to distinguish them, and I don't think that discussion should hold up this easy fix.

**Comment by argiepiano** [comment]
Thanks for tackling this! I added a small comment in the PR

**Comment by olafgrabienski** [comment]
@NormPlum Agreed, if we consider also the Views listing, it makes sense to treat the accessability aspect in a separate issue. If we do so and keep the greyed-out display for the time being, I guess we can remove the message `This display is disabled.` from this PR.

**Comment by olafgrabienski** [comment]
> I guess we can remove the message `This display is disabled.` from this PR

Sorry @NormPlum for this suggestion. I didn't realize that the message in question isn't part of your PR, so my suggestion doesn't make any sense. I guess I had seen the message in an older PR. Anyways, sorry for the noise.
